OpenRA.Mods.Common/Traits/QuantizeFacingsFromSequence.cs

```diff
@@ -0,0 +1,43 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using OpenRA.Graphics;
+using OpenRA.Mods.Common.Graphics;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	[Desc("Derive facings from sprite body sequence.")]
+	public class QuantizeFacingsFromSequenceInfo : UpgradableTraitInfo, IQuantizeBodyOrientationInfo, Requires<RenderSpritesInfo>
+	{
+		[Desc("Defines sequence to derive facings from."), SequenceReference]
+		public readonly string Sequence = "idle";
+
+		public int QuantizedBodyFacings(ActorInfo ai, SequenceProvider sequenceProvider, string race)
+		{
+			if (string.IsNullOrEmpty(Sequence))
+				throw new InvalidOperationException("Actor " + ai.Name + " is missing sequence to quantize facings from.");
+
+			var rsi = ai.Traits.Get<RenderSpritesInfo>();
+			return sequenceProvider.GetSequence(rsi.GetImage(ai, sequenceProvider, race), Sequence).Facings;
```

Comment:
> Won't this get the number of facings for the actor's "idle" sequence? (unless the default is overridden, of course)

Yes, and that's the idea since `idle` is the most common sequence and usually has the correct number of facings for that actor.